File: app/src/main/java/com/hanheldpos/utils/binding/BindingImageUtils.java
package com.hanheldpos.utils.binding;

import android.view.View;
import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.hanheldpos.R;

public class BindingImageUtils {

    @BindingAdapter("imgCircleSwitch")
    public static void setImgCircleSwitch(ImageView view, Boolean isInput) {
        if (view == null || isInput == null) return;
        int srcOutline = R.drawable.circle_outline;
        int srcSolid = R.drawable.circle_solid;

        if (isInput) {
            view.setImageResource(srcSolid);
        } else {
            view.setImageResource(srcOutline);
        }
    }

    @BindingAdapter("imgSrc")
    public static void setImgSrc(ImageView view, Integer src) {
        if (view == null) return;
        if (src == null) {
            view.setVisibility(View.GONE);
        } else {
            view.setImageResource(src);
        }
    }
}

File: app/src/main/java/com/hanheldpos/binding/BindingImageUtils.kt
package com.hanheldpos.binding

import android.content.Context
import android.view.View
import android.widget.ImageView
import androidx.core.content.ContextCompat
import androidx.databinding.BindingAdapter
import androidx.swiperefreshlayout.widget.CircularProgressDrawable
import com.bumptech.glide.Glide
import com.hanheldpos.R


@BindingAdapter("imageUrl")
    fun loadImageFromUrl(imageView: ImageView, imageUrl: String? = null) {
        imageView.apply {
            Glide.with(context)
                .load(imageUrl ?: "")
                .placeholder(startLoadingProgress(context))
                .thumbnail(0.5f)
                .error(R.drawable.bg_no_image)
                .centerCrop()
                .into(this)
        }
    }

@BindingAdapter("imgCircleSwitch")
fun setImgCircleSwitch(view: ImageView?, isInput: Boolean?) {
    if (view == null || isInput == null) return
    val srcOutline = R.drawable.circle_outline
    val srcSolid = R.drawable.circle_solid
    if (isInput) {
        view.setImageResource(srcSolid)
    } else {
        view.setImageResource(srcOutline)
    }
}

@BindingAdapter("imgSrc")
fun setImgSrc(view: ImageView?, src: Int?) {
    if (view == null) return
    if (src == null) {
        view.visibility = View.GONE
    } else {
        view.setImageResource(src)
    }
}

private fun startLoadingProgress(context: Context): CircularProgressDrawable {
    return CircularProgressDrawable(context).apply {
        strokeWidth = 4f
        centerRadius = 30f
        setColorSchemeColors(ContextCompat.getColor(context, R.color.primary))
        start()
    }
}
